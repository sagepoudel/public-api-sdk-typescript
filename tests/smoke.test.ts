import http from 'http';
import { SourcesApi } from '../api/apis';
const TEST_TOKEN = '1ba60ccaae1c1a5230';

// Keep track of the current server to close it before each test
let server = http.createServer();

async function makeServer(
    response: any,
    statusCode = 200
): Promise<SourcesApi> {
    return new Promise((resolve, reject) => {
        server = http.createServer(function (req, res) {
            expect(req as any).toMatchSnapshot({
                client: expect.any(Object),
                socket: expect.any(Object),
            });
            res.writeHead(statusCode, { 'Content-Type': 'application/json' });
            res.end(JSON.stringify(response));
        });

        server.on('error', reject);

        const port = 52338;
        server.listen(port, () => {
            const url = `http://localhost:${port}`;
            const client = new SourcesApi(url);
            client.accessToken = TEST_TOKEN;
            resolve(client);
        });
    });
}
describe('Smoke tests', () => {
    afterEach(() => {
        if (server.listening) {
            server.close();
        }
    });

    it('sends correct request and parses response for create', async () => {
        const mockedCreate = await makeServer({
            data: {
                source: {
                    id: 'source-id',
                    slug: 'my-test-source',
                    name: 'my-test-source',
                    workspaceId: 'workspace-id',
                    enabled: true,
                    writeKeys: ['write-key'],
                    metadata: {
                        id: 'IqDTy1TpoU',
                        slug: 'javascript',
                        name: 'Javascript',
                        categories: ['Website'],
                        description:
                            'This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.',
                        logos: {
                            default:
                                'https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq',
                            alt: 'https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq',
                            mark: 'https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd',
                        },
                        options: [],
                        isCloudEventSource: false,
                    },
                    settings: {},
                    labels: [],
                },
            },
        });

        const resp = await mockedCreate.createSource({
            enabled: true,
            metadataId: 'IqDTy1TpoU',
            slug: 'my-test-source',
        });

        expect(resp).toMatchSnapshot({
            response: {
                headers: {
                    date: expect.any(String),
                },
                request: expect.any(Object),
            },
        });
    });

    it('sends correct request and parses response for read', async () => {
        const mockedRead = await makeServer({
            data: {
                source: {
                    id: 'source-id',
                    slug: 'my-test-source',
                    name: 'my-test-source',
                    workspaceId: 'workspace-id',
                    enabled: true,
                    writeKeys: ['write-key'],
                    metadata: {
                        id: 'IqDTy1TpoU',
                        slug: 'javascript',
                        name: 'Javascript',
                        categories: ['Website'],
                        description:
                            'This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.',
                        logos: {
                            default:
                                'https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq',
                            alt: 'https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq',
                            mark: 'https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd',
                        },
                        options: [],
                        isCloudEventSource: false,
                    },
                    settings: {},
                    labels: [],
                },
                trackingPlanId: null,
            },
        });

        const resp = await mockedRead.getSource('source-id');

        expect(resp).toMatchSnapshot({
            response: {
                headers: {
                    date: expect.any(String),
                },
                request: expect.any(Object),
            },
        });
    });

    it('sends correct request and parses response for list', async () => {
        const mockedList = await makeServer({
            data: {
                sources: [
                    {
                        id: 'source-id',
                        slug: 'my-test-source',
                        name: 'my-test-source',
                        workspaceId: 'workspace-id',
                        enabled: true,
                        writeKeys: ['write-key'],
                        metadata: {
                            id: 'IqDTy1TpoU',
                            slug: 'javascript',
                            name: 'Javascript',
                            categories: ['Website'],
                            description:
                                'This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.',
                            logos: {
                                default:
                                    'https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq',
                                alt: 'https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq',
                                mark: 'https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd',
                            },
                            options: [],
                            isCloudEventSource: false,
                        },
                        settings: {},
                        labels: [],
                    },
                ],
                pagination: {
                    current: 'MA==',
                    next: 'MQ==',
                    totalEntries: 63,
                },
            },
        });

        const resp = await mockedList.listSources({ count: 1 });

        expect(resp).toMatchSnapshot({
            response: {
                headers: {
                    date: expect.any(String),
                },
                request: expect.any(Object),
            },
        });
    });

    it('sends correct request and parses response for update', async () => {
        const mockedUpdate = await makeServer({
            data: {
                source: {
                    id: 'source-id',
                    slug: 'my-test-source',
                    name: 'My updated source',
                    workspaceId: 'workspace-id',
                    enabled: false,
                    writeKeys: ['write-key'],
                    metadata: {
                        id: 'IqDTy1TpoU',
                        slug: 'javascript',
                        name: 'Javascript',
                        categories: ['Website'],
                        description:
                            'This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.',
                        logos: {
                            default:
                                'https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq',
                            alt: 'https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq',
                            mark: 'https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd',
                        },
                        options: [],
                        isCloudEventSource: false,
                    },
                    settings: {},
                    labels: [],
                },
            },
        });

        const resp = await mockedUpdate.updateSource('source-id', {
            name: 'My updated source',
            enabled: false,
        });

        expect(resp).toMatchSnapshot({
            response: {
                headers: {
                    date: expect.any(String),
                },
                request: expect.any(Object),
            },
        });
    });

    it('sends correct request and parses response for delete', async () => {
        const mockedDelete = await makeServer({
            data: {
                status: 'SUCCESS',
            },
        });

        const resp = await mockedDelete.deleteSource('my-source');

        expect(resp).toMatchSnapshot({
            response: {
                headers: {
                    date: expect.any(String),
                },
                request: expect.any(Object),
            },
        });
    });

    it('parses errors correctly', async () => {
        const mockedDelete = await makeServer(
            {
                errors: [
                    {
                        type: 'not-found',
                        message: "Source with id '404' not found.",
                        data: null,
                    },
                ],
            },
            404
        );

        const resp = mockedDelete.deleteSource('404');

        await expect(resp).rejects.toMatchSnapshot();
    });
});
