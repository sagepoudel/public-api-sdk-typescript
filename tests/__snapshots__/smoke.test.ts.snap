// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke tests parses errors correctly 1`] = `
{
  "_consuming": false,
  "_dumped": false,
  "_events": {
    "end": [Function],
  },
  "_eventsCount": 1,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": true,
    "awaitDrainWriters": null,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "dataEmitted": false,
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "errorEmitted": false,
    "errored": null,
    "flowing": null,
    "highWaterMark": 16384,
    "length": 0,
    "multiAwaitDrain": false,
    "needReadable": false,
    "objectMode": false,
    "pipes": [],
    "readableListening": false,
    "reading": false,
    "readingMore": true,
    "resumeScheduled": false,
    "sync": true,
    Symbol(kPaused): null,
  },
  "aborted": false,
  "client": Any<Object>,
  "complete": false,
  "httpVersion": "1.1",
  "httpVersionMajor": 1,
  "httpVersionMinor": 1,
  "method": "DELETE",
  "rawHeaders": [
    "Accept",
    "application/json",
    "Authorization",
    "Bearer 1ba60ccaae1c1a5230",
    "host",
    "localhost:52338",
    "content-length",
    "0",
    "Connection",
    "close",
  ],
  "rawTrailers": [],
  "socket": Any<Object>,
  "statusCode": null,
  "statusMessage": null,
  "upgrade": false,
  "url": "/sources/404",
  Symbol(kCapture): false,
  Symbol(kHeaders): {
    "accept": "application/json",
    "authorization": "Bearer 1ba60ccaae1c1a5230",
    "connection": "close",
    "content-length": "0",
    "host": "localhost:52338",
  },
  Symbol(kHeadersCount): 10,
  Symbol(kTrailers): null,
  Symbol(kTrailersCount): 0,
  Symbol(RequestTimeout): undefined,
}
`;

exports[`Smoke tests parses errors correctly 2`] = `[HttpError: HTTP request failed]`;

exports[`Smoke tests sends correct request and parses response for create 1`] = `
{
  "_consuming": false,
  "_dumped": false,
  "_events": {
    "end": [Function],
  },
  "_eventsCount": 1,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": true,
    "awaitDrainWriters": null,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "dataEmitted": false,
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "errorEmitted": false,
    "errored": null,
    "flowing": null,
    "highWaterMark": 16384,
    "length": 0,
    "multiAwaitDrain": false,
    "needReadable": false,
    "objectMode": false,
    "pipes": [],
    "readableListening": false,
    "reading": false,
    "readingMore": true,
    "resumeScheduled": false,
    "sync": true,
    Symbol(kPaused): null,
  },
  "aborted": false,
  "client": Any<Object>,
  "complete": false,
  "httpVersion": "1.1",
  "httpVersionMajor": 1,
  "httpVersionMinor": 1,
  "method": "POST",
  "rawHeaders": [
    "Accept",
    "application/json",
    "Authorization",
    "Bearer 1ba60ccaae1c1a5230",
    "host",
    "localhost:52338",
    "content-type",
    "application/json",
    "content-length",
    "66",
    "Connection",
    "close",
  ],
  "rawTrailers": [],
  "socket": Any<Object>,
  "statusCode": null,
  "statusMessage": null,
  "upgrade": false,
  "url": "/sources",
  Symbol(kCapture): false,
  Symbol(kHeaders): {
    "accept": "application/json",
    "authorization": "Bearer 1ba60ccaae1c1a5230",
    "connection": "close",
    "content-length": "66",
    "content-type": "application/json",
    "host": "localhost:52338",
  },
  Symbol(kHeadersCount): 12,
  Symbol(kTrailers): null,
  Symbol(kTrailersCount): 0,
  Symbol(RequestTimeout): undefined,
}
`;

exports[`Smoke tests sends correct request and parses response for create 2`] = `
{
  "body": CreateSource200Response {
    "data": CreateSourceV1Output {
      "source": Source5 {
        "enabled": true,
        "id": "source-id",
        "labels": [],
        "metadata": Metadata1 {
          "categories": [
            "Website",
          ],
          "description": "This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.",
          "id": "IqDTy1TpoU",
          "isCloudEventSource": false,
          "logos": Logos1 {
            "alt": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
            "default": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
            "mark": "https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd",
          },
          "name": "Javascript",
          "options": [],
          "slug": "javascript",
        },
        "name": "my-test-source",
        "settings": {},
        "slug": "my-test-source",
        "workspaceId": "workspace-id",
        "writeKeys": [
          "write-key",
        ],
      },
    },
  },
  "response": {
    "body": {
      "data": {
        "source": {
          "enabled": true,
          "id": "source-id",
          "labels": [],
          "metadata": {
            "categories": [
              "Website",
            ],
            "description": "This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.",
            "id": "IqDTy1TpoU",
            "isCloudEventSource": false,
            "logos": {
              "alt": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
              "default": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
              "mark": "https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd",
            },
            "name": "Javascript",
            "options": [],
            "slug": "javascript",
          },
          "name": "my-test-source",
          "settings": {},
          "slug": "my-test-source",
          "workspaceId": "workspace-id",
          "writeKeys": [
            "write-key",
          ],
        },
      },
    },
    "headers": {
      "connection": "close",
      "content-type": "application/json",
      "date": Any<String>,
      "transfer-encoding": "chunked",
    },
    "request": {
      "headers": undefined,
      "method": undefined,
      "uri": undefined,
    },
    "statusCode": 200,
  },
}
`;

exports[`Smoke tests sends correct request and parses response for delete 1`] = `
{
  "_consuming": false,
  "_dumped": false,
  "_events": {
    "end": [Function],
  },
  "_eventsCount": 1,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": true,
    "awaitDrainWriters": null,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "dataEmitted": false,
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "errorEmitted": false,
    "errored": null,
    "flowing": null,
    "highWaterMark": 16384,
    "length": 0,
    "multiAwaitDrain": false,
    "needReadable": false,
    "objectMode": false,
    "pipes": [],
    "readableListening": false,
    "reading": false,
    "readingMore": true,
    "resumeScheduled": false,
    "sync": true,
    Symbol(kPaused): null,
  },
  "aborted": false,
  "client": Any<Object>,
  "complete": false,
  "httpVersion": "1.1",
  "httpVersionMajor": 1,
  "httpVersionMinor": 1,
  "method": "DELETE",
  "rawHeaders": [
    "Accept",
    "application/json",
    "Authorization",
    "Bearer 1ba60ccaae1c1a5230",
    "host",
    "localhost:52338",
    "content-length",
    "0",
    "Connection",
    "close",
  ],
  "rawTrailers": [],
  "socket": Any<Object>,
  "statusCode": null,
  "statusMessage": null,
  "upgrade": false,
  "url": "/sources/my-source",
  Symbol(kCapture): false,
  Symbol(kHeaders): {
    "accept": "application/json",
    "authorization": "Bearer 1ba60ccaae1c1a5230",
    "connection": "close",
    "content-length": "0",
    "host": "localhost:52338",
  },
  Symbol(kHeadersCount): 10,
  Symbol(kTrailers): null,
  Symbol(kTrailersCount): 0,
  Symbol(RequestTimeout): undefined,
}
`;

exports[`Smoke tests sends correct request and parses response for delete 2`] = `
{
  "body": DeleteSource200Response {
    "data": DeleteSourceV1Output {
      "status": "SUCCESS",
    },
  },
  "response": {
    "body": {
      "data": {
        "status": "SUCCESS",
      },
    },
    "headers": {
      "connection": "close",
      "content-type": "application/json",
      "date": Any<String>,
      "transfer-encoding": "chunked",
    },
    "request": {
      "headers": undefined,
      "method": undefined,
      "uri": undefined,
    },
    "statusCode": 200,
  },
}
`;

exports[`Smoke tests sends correct request and parses response for list 1`] = `
{
  "_consuming": false,
  "_dumped": false,
  "_events": {
    "end": [Function],
  },
  "_eventsCount": 1,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": true,
    "awaitDrainWriters": null,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "dataEmitted": false,
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "errorEmitted": false,
    "errored": null,
    "flowing": null,
    "highWaterMark": 16384,
    "length": 0,
    "multiAwaitDrain": false,
    "needReadable": false,
    "objectMode": false,
    "pipes": [],
    "readableListening": false,
    "reading": false,
    "readingMore": true,
    "resumeScheduled": false,
    "sync": true,
    Symbol(kPaused): null,
  },
  "aborted": false,
  "client": Any<Object>,
  "complete": false,
  "httpVersion": "1.1",
  "httpVersionMajor": 1,
  "httpVersionMinor": 1,
  "method": "GET",
  "rawHeaders": [
    "Accept",
    "application/json",
    "Authorization",
    "Bearer 1ba60ccaae1c1a5230",
    "host",
    "localhost:52338",
    "Connection",
    "close",
  ],
  "rawTrailers": [],
  "socket": Any<Object>,
  "statusCode": null,
  "statusMessage": null,
  "upgrade": false,
  "url": "/sources?pagination%5Bcount%5D=1",
  Symbol(kCapture): false,
  Symbol(kHeaders): {
    "accept": "application/json",
    "authorization": "Bearer 1ba60ccaae1c1a5230",
    "connection": "close",
    "host": "localhost:52338",
  },
  Symbol(kHeadersCount): 8,
  Symbol(kTrailers): null,
  Symbol(kTrailersCount): 0,
  Symbol(RequestTimeout): undefined,
}
`;

exports[`Smoke tests sends correct request and parses response for list 2`] = `
{
  "body": ListSources200Response {
    "data": ListSourcesV1Output {
      "pagination": Pagination {
        "current": "MA==",
        "next": "MQ==",
        "previous": undefined,
        "totalEntries": 63,
      },
      "sources": [
        SourceV1 {
          "enabled": true,
          "id": "source-id",
          "labels": [],
          "metadata": Metadata1 {
            "categories": [
              "Website",
            ],
            "description": "This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.",
            "id": "IqDTy1TpoU",
            "isCloudEventSource": false,
            "logos": Logos1 {
              "alt": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
              "default": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
              "mark": "https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd",
            },
            "name": "Javascript",
            "options": [],
            "slug": "javascript",
          },
          "name": "my-test-source",
          "settings": {},
          "slug": "my-test-source",
          "workspaceId": "workspace-id",
          "writeKeys": [
            "write-key",
          ],
        },
      ],
    },
  },
  "response": {
    "body": {
      "data": {
        "pagination": {
          "current": "MA==",
          "next": "MQ==",
          "totalEntries": 63,
        },
        "sources": [
          {
            "enabled": true,
            "id": "source-id",
            "labels": [],
            "metadata": {
              "categories": [
                "Website",
              ],
              "description": "This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.",
              "id": "IqDTy1TpoU",
              "isCloudEventSource": false,
              "logos": {
                "alt": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
                "default": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
                "mark": "https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd",
              },
              "name": "Javascript",
              "options": [],
              "slug": "javascript",
            },
            "name": "my-test-source",
            "settings": {},
            "slug": "my-test-source",
            "workspaceId": "workspace-id",
            "writeKeys": [
              "write-key",
            ],
          },
        ],
      },
    },
    "headers": {
      "connection": "close",
      "content-type": "application/json",
      "date": Any<String>,
      "transfer-encoding": "chunked",
    },
    "request": {
      "headers": undefined,
      "method": undefined,
      "uri": undefined,
    },
    "statusCode": 200,
  },
}
`;

exports[`Smoke tests sends correct request and parses response for read 1`] = `
{
  "_consuming": false,
  "_dumped": false,
  "_events": {
    "end": [Function],
  },
  "_eventsCount": 1,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": true,
    "awaitDrainWriters": null,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "dataEmitted": false,
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "errorEmitted": false,
    "errored": null,
    "flowing": null,
    "highWaterMark": 16384,
    "length": 0,
    "multiAwaitDrain": false,
    "needReadable": false,
    "objectMode": false,
    "pipes": [],
    "readableListening": false,
    "reading": false,
    "readingMore": true,
    "resumeScheduled": false,
    "sync": true,
    Symbol(kPaused): null,
  },
  "aborted": false,
  "client": Any<Object>,
  "complete": false,
  "httpVersion": "1.1",
  "httpVersionMajor": 1,
  "httpVersionMinor": 1,
  "method": "GET",
  "rawHeaders": [
    "Accept",
    "application/json",
    "Authorization",
    "Bearer 1ba60ccaae1c1a5230",
    "host",
    "localhost:52338",
    "Connection",
    "close",
  ],
  "rawTrailers": [],
  "socket": Any<Object>,
  "statusCode": null,
  "statusMessage": null,
  "upgrade": false,
  "url": "/sources/source-id",
  Symbol(kCapture): false,
  Symbol(kHeaders): {
    "accept": "application/json",
    "authorization": "Bearer 1ba60ccaae1c1a5230",
    "connection": "close",
    "host": "localhost:52338",
  },
  Symbol(kHeadersCount): 8,
  Symbol(kTrailers): null,
  Symbol(kTrailersCount): 0,
  Symbol(RequestTimeout): undefined,
}
`;

exports[`Smoke tests sends correct request and parses response for read 2`] = `
{
  "body": GetSource200Response {
    "data": GetSourceV1Output {
      "source": Source4 {
        "enabled": true,
        "id": "source-id",
        "labels": [],
        "metadata": Metadata1 {
          "categories": [
            "Website",
          ],
          "description": "This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.",
          "id": "IqDTy1TpoU",
          "isCloudEventSource": false,
          "logos": Logos1 {
            "alt": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
            "default": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
            "mark": "https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd",
          },
          "name": "Javascript",
          "options": [],
          "slug": "javascript",
        },
        "name": "my-test-source",
        "settings": {},
        "slug": "my-test-source",
        "workspaceId": "workspace-id",
        "writeKeys": [
          "write-key",
        ],
      },
    },
  },
  "response": {
    "body": {
      "data": {
        "source": {
          "enabled": true,
          "id": "source-id",
          "labels": [],
          "metadata": {
            "categories": [
              "Website",
            ],
            "description": "This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.",
            "id": "IqDTy1TpoU",
            "isCloudEventSource": false,
            "logos": {
              "alt": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
              "default": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
              "mark": "https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd",
            },
            "name": "Javascript",
            "options": [],
            "slug": "javascript",
          },
          "name": "my-test-source",
          "settings": {},
          "slug": "my-test-source",
          "workspaceId": "workspace-id",
          "writeKeys": [
            "write-key",
          ],
        },
      },
    },
    "headers": {
      "connection": "close",
      "content-type": "application/json",
      "date": Any<String>,
      "transfer-encoding": "chunked",
    },
    "request": {
      "headers": undefined,
      "method": undefined,
      "uri": undefined,
    },
    "statusCode": 200,
  },
}
`;

exports[`Smoke tests sends correct request and parses response for update 1`] = `
{
  "_consuming": false,
  "_dumped": false,
  "_events": {
    "end": [Function],
  },
  "_eventsCount": 1,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": true,
    "awaitDrainWriters": null,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "closeEmitted": false,
    "closed": false,
    "constructed": true,
    "dataEmitted": false,
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "errorEmitted": false,
    "errored": null,
    "flowing": null,
    "highWaterMark": 16384,
    "length": 0,
    "multiAwaitDrain": false,
    "needReadable": false,
    "objectMode": false,
    "pipes": [],
    "readableListening": false,
    "reading": false,
    "readingMore": true,
    "resumeScheduled": false,
    "sync": true,
    Symbol(kPaused): null,
  },
  "aborted": false,
  "client": Any<Object>,
  "complete": false,
  "httpVersion": "1.1",
  "httpVersionMajor": 1,
  "httpVersionMinor": 1,
  "method": "PATCH",
  "rawHeaders": [
    "Accept",
    "application/json",
    "Authorization",
    "Bearer 1ba60ccaae1c1a5230",
    "host",
    "localhost:52338",
    "content-type",
    "application/json",
    "content-length",
    "44",
    "Connection",
    "close",
  ],
  "rawTrailers": [],
  "socket": Any<Object>,
  "statusCode": null,
  "statusMessage": null,
  "upgrade": false,
  "url": "/sources/source-id",
  Symbol(kCapture): false,
  Symbol(kHeaders): {
    "accept": "application/json",
    "authorization": "Bearer 1ba60ccaae1c1a5230",
    "connection": "close",
    "content-length": "44",
    "content-type": "application/json",
    "host": "localhost:52338",
  },
  Symbol(kHeadersCount): 12,
  Symbol(kTrailers): null,
  Symbol(kTrailersCount): 0,
  Symbol(RequestTimeout): undefined,
}
`;

exports[`Smoke tests sends correct request and parses response for update 2`] = `
{
  "body": UpdateSource200Response {
    "data": UpdateSourceV1Output {
      "source": Source6 {
        "enabled": false,
        "id": "source-id",
        "labels": [],
        "metadata": Metadata1 {
          "categories": [
            "Website",
          ],
          "description": "This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.",
          "id": "IqDTy1TpoU",
          "isCloudEventSource": false,
          "logos": Logos1 {
            "alt": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
            "default": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
            "mark": "https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd",
          },
          "name": "Javascript",
          "options": [],
          "slug": "javascript",
        },
        "name": "My updated source",
        "settings": {},
        "slug": "my-test-source",
        "workspaceId": "workspace-id",
        "writeKeys": [
          "write-key",
        ],
      },
    },
  },
  "response": {
    "body": {
      "data": {
        "source": {
          "enabled": false,
          "id": "source-id",
          "labels": [],
          "metadata": {
            "categories": [
              "Website",
            ],
            "description": "This is our most flexible and powerful tracking system, using analytics.js.  Track and analyze information about your visitors and customers, and every action that they take, in any of our 140 integrations, business intelligence tools, or directly with SQL tools.",
            "id": "IqDTy1TpoU",
            "isCloudEventSource": false,
            "logos": {
              "alt": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
              "default": "https://cdn.filepicker.io/api/file/aRgo4XJQZausZxD4gZQq",
              "mark": "https://cdn.filepicker.io/api/file/kBpmEoSSaakidAvoFmzd",
            },
            "name": "Javascript",
            "options": [],
            "slug": "javascript",
          },
          "name": "My updated source",
          "settings": {},
          "slug": "my-test-source",
          "workspaceId": "workspace-id",
          "writeKeys": [
            "write-key",
          ],
        },
      },
    },
    "headers": {
      "connection": "close",
      "content-type": "application/json",
      "date": Any<String>,
      "transfer-encoding": "chunked",
    },
    "request": {
      "headers": undefined,
      "method": undefined,
      "uri": undefined,
    },
    "statusCode": 200,
  },
}
`;
